sycl/include/sycl/ext/oneapi/experimental/annotated_ptr/annotated_ptr.hpp

```diff
@@ -238,6 +238,11 @@ __SYCL_TYPE(annotated_ptr) annotated_ptr<T, detail::properties_t<Props...>> {
   annotated_ptr(const annotated_ptr &) = default;
   annotated_ptr &operator=(const annotated_ptr &) = default;
 
+  static_assert(
+      std::is_same_v<T, void> || std::is_trivially_copyable_v<T>,
+      "encapsulating annotated_ptr with non-trivially-copyable type is "
+      "not supported!");
```

Comment:
Nit: It's a weird place to have a sudden static_assert. I would prefer we have it at the beginning of the class.

Also, while we are here:
```suggestion
  static_assert(
      std::is_same_v<T, void> || std::is_trivially_copyable_v<T>,
      "Encapsulating annotated_ptr with non-trivially-copyable type is "
      "not supported!");
```